js/liqui.js

```diff
@@ -289,22 +289,26 @@ module.exports = class liqui extends Exchange {
         let symbol = undefined;
         if (market)
             symbol = market['symbol'];
+        let amount = trade['amount'];
         let feeSide = (side == 'buy') ? 'base' : 'quote';
+        let type = 'limit';
+        if (trade['is_your_order']) { // this is filled by fetchMyTrades() only
```

Comment:
We will need to check if the `is_your_order` key is there before accessing it, if it disappears. I know, this is not intuitive in JS, but Python is indeed very sensitive to everything (which is a little bit of pain, on the other side, this may as well be your advantage, especially when you're dealing with unreliable systems).

Reply:
Oh, that's easy: `'amount'` is always there. So, when we are sure, it's there always, we can reference it right away, but it the key is optional, then we have to check for it first, because it has no default value and throws an exception in Python and a warning in PHP ) Does this clarify it somewhat?

Reply:
Well... but how transpiler knows about it?

Reply:
That's the point, it doesn't know anything about it, because the transpiler works in *compile-time* not in *run-time*, so the only thing a transpiler can do, is just convert that line as is to the other language. And it transpiles perfectly well. But throws an exception later in run-time, so, the transpiler is not involved into the above particular case.

Reply:
Ok, I've got it.
It will be pretty hard for me to keep these might-be-missing parameters in mind, although I'll try all my best to do.